File: mobile/iosApp/iosApp/ConnectionStatusBar.swift
import SwiftUI

/// Compact connection status bar component for iOS.
/// Displays an 8pt dot + label indicating current connection state.
struct ConnectionStatusBar: View {
    let connectionState: ConnectionState
    let onTap: () -> Void

    @State private var pulseOpacity: Double = 1.0

    var body: some View {
        Button(action: onTap) {
            HStack(spacing: 8) {
                // Status indicator dot (8pt)
                Circle()
                    .fill(statusColor)
                    .frame(width: 8, height: 8)
                    .opacity(pulseOpacity)
                    .onAppear {
                        if case .reconnecting = connectionState {
                            startPulseAnimation()
                        }
                    }
                    .animation(.easeInOut(duration: 0.2), value: statusColor)

                // Status text label
                Text(statusText)
                    .font(TinsuTypography.label)
                    .foregroundColor(TinsuColors.onSurface)
            }
            .padding(.horizontal, 12)
            .padding(.vertical, 8)
        }
        .buttonStyle(PlainButtonStyle())
    }

    private var statusColor: Color {
        switch connectionState {
        case .connected:
            return TinsuColors.success
        case .reconnecting:
            return TinsuColors.warning
        case .disconnected:
            return TinsuColors.error
        case .offline:
            return TinsuColors.outline
        }
    }

    private var statusText: String {
        switch connectionState {
        case .connected:
            return "Connected"
        case .reconnecting:
            return "Reconnecting..."
        case .disconnected:
            return "Disconnected"
        case .offline:
            return "Offline"
        }
    }

    private func startPulseAnimation() {
        withAnimation(
            Animation
                .easeInOut(duration: 1.0)
                .repeatForever(autoreverses: true)
        ) {
            pulseOpacity = 0.5
        }
    }
}

/// Connection state enum for iOS UI components.
enum ConnectionState {
    case connected(host: String, port: Int, transport: String, uptime: TimeInterval)
    case reconnecting
    case disconnected(reason: String?)
    case offline
}

#Preview {
    VStack(spacing: 16) {
        ConnectionStatusBar(connectionState: .connected(host: "192.168.1.100", port: 22, transport: "SSH", uptime: 3665)) {}
        ConnectionStatusBar(connectionState: .reconnecting) {}
        ConnectionStatusBar(connectionState: .disconnected(reason: nil)) {}
        ConnectionStatusBar(connectionState: .offline) {}
    }
    .padding()
    .background(TinsuColors.background)
}

File: mobile/androidApp/src/main/java/com/tinsu/mobile/ui/connection/ConnectionStatusBar.kt
package com.tinsu.mobile.ui.connection

import androidx.compose.animation.animateColorAsState
import androidx.compose.animation.core.animateFloatAsState
import androidx.compose.animation.core.infiniteRepeatable
import androidx.compose.animation.core.rewind
import androidx.compose.animation.core.tween
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import com.tinsu.mobile.connection.ConnectionEvent

/**
 * Compact connection status bar component.
 * Displays an 8dp dot + label indicating current connection state.
 *
 * @param connectionState The current connection event/state
 * @param onClick Callback when the status bar is tapped
 * @param modifier Modifier for the component
 */
@Composable
fun ConnectionStatusBar(
    connectionState: ConnectionEvent,
    onClick: () -> Unit,
    modifier: Modifier = Modifier
) {
    val (statusText, statusColor) = when (connectionState) {
        is ConnectionEvent.Connected -> {
            Pair("Connected", MaterialTheme.colorScheme.tinsuColors.success)
        }
        is ConnectionEvent.Reconnecting -> {
            Pair("Reconnecting...", MaterialTheme.colorScheme.tinsuColors.warning)
        }
        is ConnectionEvent.Disconnected -> {
            Pair("Disconnected", MaterialTheme.colorScheme.error)
        }
        is ConnectionEvent.Offline -> {
            Pair("Offline", MaterialTheme.colorScheme.outline)
        }
    }

    // Animate color transition with 200ms fade
    val animatedColor by animateColorAsState(
        targetValue = statusColor,
        animationSpec = tween(durationMillis = 200),
        label = "statusColor"
    )

    // Pulse animation for Reconnecting state
    val pulseAlpha by animateFloatAsState(
        targetValue = if (connectionState is ConnectionEvent.Reconnecting) 0.5f else 1.0f,
        animationSpec = if (connectionState is ConnectionEvent.Reconnecting) {
            infiniteRepeatable(
                animation = tween(durationMillis = 1000),
                repeatMode = androidx.compose.animation.core.RepeatMode.Reverse
            )
        } else {
            tween(durationMillis = 200)
        },
        label = "pulseAlpha"
    )

    Row(
        modifier = modifier
            .clickable(onClick = onClick)
            .padding(horizontal = 12.dp, vertical = 8.dp),
        horizontalArrangement = Arrangement.spacedBy(8.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        // Status indicator dot (8dp)
        Box(
            modifier = Modifier
                .size(8.dp)
                .background(animatedColor, CircleShape)
                .alpha(pulseAlpha)
        )

        // Status text label
        Text(
            text = statusText,
            style = MaterialTheme.typography.labelMedium,
            color = MaterialTheme.colorScheme.onSurface
        )
    }
}
